src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+public abstract class Figure implements AreaCalculationInterface, FigureDataInterface {
+    protected String figureName;
+    protected String color;
+
+    public Figure(String color, String figureName) {
+        this.color = color;
+        this.figureName = figureName;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    public void draw() {
+        System.out.println("Figure: " + getFigureName() + " area: "
+                + getArea() + "sq. units, side:"
```

Comment:
The string concatenation in 'getFigureData()' suggests that this method may be returning a String representation of figure data, which might not be the best practice. It's better to have a method that returns the actual data in a more structured way.